src/roly_poly.c

```diff
@@ -0,0 +1,462 @@
+#include "roly_poly.h"
+#include "kirby.h"
+#include "random.h"
+#include "task.h"
+
+static void sub_080ACDA4(struct Object2*);
+static void RolyPolyIdle(struct Object2*);
+static void RolyPolyJump(struct Object2*);
+static void RolyPolyJumping(struct Object2*);
+static void RolyPolyRollLeadIn(struct Object2*);
+static void RolyPolyRollingRight(struct Object2*);
+static void RolyPolyRollingLeft(struct Object2*);
+static void sub_080AD5D4(struct Object2*);
+static void RolyPolyRoll(struct Object2*);
+static void RolyPolyRolling(struct Object2*);
+
+void* CreateRolyPoly(struct Object* arg0, u8 playerId) {
+    struct Task* task = TaskCreate(ObjectMain, sizeof(struct Object2), 0x1000, TASK_USE_EWRAM, ObjectDestroy);
+    struct Object2* tmp = TaskGetStructPtr(task);
+    struct Object2* rolypoly = tmp;
+
+    InitObject(rolypoly, arg0, playerId);
+    if (rolypoly->base.x > rolypoly->kirby3->base.base.base.x) {
+        rolypoly->base.flags |= 0x01;
+    }
+    else {
+        rolypoly->base.flags &= ~0x01;
+    }
+    sub_0803E2B0(&rolypoly->base, -5, -4, 5, 7);
+    sub_0803E308(&rolypoly->base, -6, -5, 6, 9);
+    ObjectInitSprite(rolypoly);
+    gUnk_08351648[rolypoly->type].unk10(rolypoly);
+    rolypoly->unk9E = 0;
+    rolypoly->unk7C = sub_080ACDA4;
+    return rolypoly;
+}
+
+// Always run, always stored in rolypoly->unk7C
+// -> Background-Method?
+static void sub_080ACDA4(struct Object2* rolypoly) {
+    u8 oldUnk9D;
+    u8 newUnk9D = 0;
+
+    if (!Macro_0810B1F4(&rolypoly->base)) {
+        if ((rolypoly->base.counter & 0x1f) == 0x1f) {
+            rolypoly->kirby3 = sub_0803D368(&rolypoly->base);
+        }
+
+        switch (rolypoly->unk83) {
+            case 0:
```

Comment:
Let's remove the indent for switch cases. At least this part is consistent throughout the project :)